File: queen/src/main/java/com/scatum/queen/Piece.java
package com.scatum.queen;


public enum Piece {

    BLACK_KING("bK", Color.BLACK),
    BLACK_QUEEN("bQ", Color.BLACK),
    BLACK_ROOK("bR", Color.BLACK),
    BLACK_BISHOP("bB", Color.BLACK),
    BLACK_HORSE("bH", Color.BLACK),
    BLACK_PAWN("bP", Color.BLACK),

    WHITE_KING("wK", Color.WHITE),
    WHITE_QUEEN("wQ", Color.WHITE),
    WHITE_ROOK("wR", Color.WHITE),
    WHITE_BISHOP("wB", Color.WHITE),
    WHITE_HORSE("wH", Color.WHITE),
    WHITE_PAWN("wP", Color.WHITE),

    EMPTY("..", Color.NONE),
    BLACK_ATTACK_ZONE("--", Color.NONE),
    WHITE_ATTACK_ZONE("++", Color.NONE);

    enum Color {
        BLACK,
        WHITE,
        NONE;
    }

    private String value;
    private Color color;

    private Piece(String value, Color color) {
        this.value = value;
        this.color = color;
    }

    public String value() {
        return value;
    }

    public Color color() {
        return color;
    }

    public String attackingZone() {
        if (this.color == Color.WHITE) {
            return Piece.WHITE_ATTACK_ZONE.value();
        }
        if (this.color == Color.BLACK) {
            return Piece.BLACK_ATTACK_ZONE.value();
        }

        return null;
    }

    public String opponentAttackingZone() {
        if (this.color == Color.WHITE) {
            return Piece.BLACK_ATTACK_ZONE.value();
        }
        if (this.color == Color.BLACK) {
            return Piece.WHITE_ATTACK_ZONE.value();
        }

        return null;
    }
}

File: queen-csharp/Piece.cs
using System;

namespace QueenApplication
{
    class Piece
    {
        public enum Color { BLACK, WHITE, NONE }

        public static readonly Piece BLACK_KING = new Piece("bK", Color.BLACK);
        public static readonly Piece BLACK_QUEEN = new Piece("bQ", Color.BLACK);
        public static readonly Piece BLACK_ROOK = new Piece("bR", Color.BLACK);
        public static readonly Piece BLACK_BISHOP = new Piece("bB", Color.BLACK);
        public static readonly Piece BLACK_HORSE = new Piece("bH", Color.BLACK);
        public static readonly Piece BLACK_PAWN = new Piece("bP", Color.BLACK);

        public static readonly Piece WHITE_KING = new Piece("wK", Color.WHITE);
        public static readonly Piece WHITE_QUEEN = new Piece("wQ", Color.WHITE);
        public static readonly Piece WHITE_ROOK = new Piece("wR", Color.WHITE);
        public static readonly Piece WHITE_BISHOP = new Piece("wB", Color.WHITE);
        public static readonly Piece WHITE_HORSE = new Piece("wH", Color.WHITE);
        public static readonly Piece WHITE_PAWN = new Piece("wP", Color.WHITE);

        public static readonly Piece EMPTY = new Piece("..", Color.NONE);
        public static readonly Piece BLACK_ATTACK_ZONE = new Piece("--", Color.NONE);
        public static readonly Piece WHITE_ATTACK_ZONE = new Piece("++", Color.NONE);


        private readonly String value;
        private readonly Color color;

        Piece(String value, Color color)
        {
            this.value = value;
            this.color = color;
        }

        public String Value()
        {
            return value;
        }

        public Color GetColor()
        {
            return color;
        }

        public String AttackingZone()
        {
            if (this.color == Color.WHITE)
            {
                return WHITE_ATTACK_ZONE.Value();
            }
            if (this.color == Color.BLACK)
            {
                return BLACK_ATTACK_ZONE.Value();
            }

            return null;
        }

        public String OpponentAttackingZone()
        {
            if (this.color == Color.WHITE)
            {
                return BLACK_ATTACK_ZONE.Value();
            }
            if (this.color == Color.BLACK)
            {
                return WHITE_ATTACK_ZONE.Value();
            }

            return null;
        }
    }
}
